source/common/quic/envoy_quic_server_stream.cc

```diff
@@ -34,11 +34,12 @@ EnvoyQuicServerStream::EnvoyQuicServerStream(
           static_cast<uint32_t>(GetReceiveWindow().value()), *filterManagerConnection(),
           [this]() { runLowWatermarkCallbacks(); }, [this]() { runHighWatermarkCallbacks(); },
           stats, http3_options),
-      headers_with_underscores_action_(headers_with_underscores_action) {
+      headers_with_underscores_action_(headers_with_underscores_action), session_(session),
+      capsule_protocol_handler_(this) {
   ASSERT(static_cast<uint32_t>(GetReceiveWindow().value()) > 8 * 1024,
          "Send buffer limit should be larger than 8KB.");
 
-  stats_gatherer_ = new QuicStatsGatherer(&connection()->dispatcher().timeSource());
+  stats_gatherer_ = new QuicStatsGatherer(&filterManagerConnection()->dispatcher().timeSource());
```

Comment:
Out of curiosity, why this change?

Reply:
I guess we do not need a runtime flag anymore after the change you've made. Let me know if you think we still need one.

Reply:
clangtidy doesn't allow calling a virtual function in a constructor.